### PR TITLE
feat(bpdm-cert): created metadata endpoints for certificate types

### LINKED ISSUE #6: API: MetadataController / Management API
As a developer that implements the business partner certificate API I want to get a list of registered certificate types so that I know how the spelling is and what certificate types are available - that's needed for the usage of other API endpoints in this use case.

As a developer that implements the business partner certificate API I want to register a new certificate type, so that I can provide new certificates where the type is not known so far. That's needed for the post API of this use case.

API-Call (get): MetaDataController/api/catena/certificate-types - returns all currently registered certificate types
API-Call (post): MetaDataController/api/catena/create-certificate-type - register certificate type

### PR DESCRIPTION
**Pull request by SujitMBRDI**
## Description
For Bpdm Certificate Management application, with this pull request created two endpoints to create and retrive certificate types under Metadata controller.

Fixes #6 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by nicoprow — COMMENTED** [review]
MetadataController works as expected.

Just a general feedback for the application setup:
Currently when compiling the maven project, it generates source files for the whole project. But these source files are nowhere used, making it unnecessary to be created. Since created source files there are in the same package, they cause redeclaration errors in the IDE. 
I see 3 options to optimize:
1. Only generate source files that you need and refer to them in the actual project (don't copy past them). In this way, creating source files is meaninful.
2. Make a special maven goal for generating the source files, so that they not always generate when compiling the project.
3. Generate the source files in it's own package so that they don't interfere with the actual project's files.

I think option 1 would be the most integrated one.

**Review by nicoprow — APPROVED** [review]

**Comment by SujitMBRDI** [comment]
> MetadataController works as expected.
> 
> Just a general feedback for the application setup: Currently when compiling the maven project, it generates source files for the whole project. But these source files are nowhere used, making it unnecessary to be created. Since created source files there are in the same package, they cause redeclaration errors in the IDE. I see 3 options to optimize:
> 
> 1. Only generate source files that you need and refer to them in the actual project (don't copy past them). In this way, creating source files is meaninful.
> 2. Make a special maven goal for generating the source files, so that they not always generate when compiling the project.
> 3. Generate the source files in it's own package so that they don't interfere with the actual project's files.
> 
> I think option 1 would be the most integrated one.

Yeah, I have adapted changes.

File: src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/controller/MetadataApi.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    )
+    @PostMapping("/certificate-types")
+    @PostExchange("/certificate-types")
+    fun setCertificateType(@Parameter(description = "", required = true) @Valid @RequestBody certificateTypeDto: CertificateTypeDto): CertificateTypeDto
+
+
+    @Operation(
+        summary = "Get certificate types",
+        operationId = "getCertificateTypes",
+        description = """Get a list of all currently registered certificate types.""",
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/controller/MetadataApi.kt]: Removed

**Comment by SujitMBRDI** [comment]
> > MetadataController works as expected.
> > Just a general feedback for the application setup: Currently when compiling the maven project, it generates source files for the whole project. But these source files are nowhere used, making it unnecessary to be created. Since created source files there are in the same package, they cause redeclaration errors in the IDE. I see 3 options to optimize:
> > 
> > 1. Only generate source files that you need and refer to them in the actual project (don't copy past them). In this way, creating source files is meaninful.
> > 2. Make a special maven goal for generating the source files, so that they not always generate when compiling the project.
> > 3. Generate the source files in it's own package so that they don't interfere with the actual project's files.
> > 
> > I think option 1 would be the most integrated one.
> 
> Yeah, I have adapted changes.


@nicoprow, I know you approved the changes before but i have made some more changes related to docker-compose file which i have added new so kindly review again and let me know your feedback on same.

**Review by nicoprow — APPROVED** [review]
Looks good to me

**Pull request merged**